private/lib/Service/CategoryService.php

```diff
@@ -82,32 +122,59 @@ public function updateCategory(int $userId, int $categoryId, string $categoryNam
         $this->repository->save();
     }
 
-    public function deleteCategoryAndAssociatedEntries(int $categoryId, int $userId): void
+    /**
+     * @throws InvalidArgumentException
+     * @throws ORMException
+     * @throws NotFoundException
+     */
+    public function deleteCategory(int $userId, int $categoryId): void
     {
         $category = $this->categoryHelper->getCategoryForUser($categoryId, $userId);
+
+        $this->setUncategorizedEntriesAndTemplates($userId, $category);
+
+        // queue category for deleting
+        $this->repository->remove($category);
+
+        // delete queued entries, templates and categories
+        $this->repository->save();
+    }
+
+    /**
+     * @throws NotFoundException
+     * @throws InvalidArgumentException
+     */
+    public function setUncategorizedEntriesAndTemplates(int $userId, Category $category)
+    {
+        $categoryId = $category->getId();
         $user = $this->userHelper->getUserById($userId);
 
-        // get associated entries and queue for deleting
-        $entries = $this->entryHelper->getEntriesForUserByCategoryId($userId, $categoryId);
-        foreach ($entries as $entry) {
-            $this->repository->remove($entry);
+        // Creates uncategorized category if user doesn't have one
+        $uncategorizedCategory = $this->categoryHelper->getCategoryByUserAndCategoryName($user, Category::UNCATEGORIZED_CATEGORY_NAME);
+        if ($uncategorizedCategory === null) {
+            $uncategorizedCategory = $this->createCategory($userId, Category::UNCATEGORIZED_CATEGORY_NAME, Category::UNCATEGORIZED_CATEGORY_DESCRIPTION);
+
+            // Set default sort order to 0
+            $uncategorizedCategory->setSortOrder(0);
```

Comment:
Could you remove this? And set the default value of setSortOrder to 0 via the category model?

Reply:
you still didnt remove this?